extra_foam/gui/image_tool/reference_view.py

```diff
@@ -65,6 +76,22 @@ def updateF(self, data, auto_update):
             # is deferred.
             self._reference.setImage(data.image.reference)
 
+        if self._is_recording:
+            # Update cached image and count
+            image = self._corrected.image
+            if self._recorded_image is None:
+                self._recorded_image = image
+            else:
+                # Average without storing images
```

Comment:
Is it possible to use the [`RawImageData`](https://github.com/European-XFEL/EXtra-foam/blob/dev/extra_foam/pipeline/data_model.py#L152) class here? It already supports keeping a moving average, and it uses the C++ implementation of the moving average if possible.

Reply:
I tried and I realized that the `MovingAverageArray` classes are meant to be used with the `_BaseProcessor` variants. These are enabled with the `MetaProcessor` metaclass. Prolly it's not worth to reimplement things for this relatively small code change.

We could revisit again when there are more relevant changes :D

Reply:
Ahhh I see. Maybe we could use `extra_foam.algorithms.movingAvgImageData()` then? That's implemented in C++.

Reply:
Done! Thanks for the suggestion :D